packages/lodestar/src/eth1/impl/ethers.ts

```diff
@@ -129,10 +129,34 @@ export class EthersEth1Notifier extends (EventEmitter as { new(): Eth1EventEmitt
     }
   }
 
+  /**
+   * Process blocks from lastProcessedEth1BlockNumber + 1 until toNumber.
+   * @param toNumber
+   */
   public async processBlocks(toNumber: number): Promise<void> {
-    let blockNumber = this.lastProcessedEth1BlockNumber + 1;
-    while (blockNumber <= toNumber && await this.processBlock(blockNumber)) {
-      blockNumber++;
+    let rangeBlockNumber = this.lastProcessedEth1BlockNumber;
+    while (rangeBlockNumber < toNumber && this.started) {
+      const blockNumber = Math.min(this.lastProcessedEth1BlockNumber + 100, toNumber);
+      let rangeDepositEvents;
+      try {
+        rangeDepositEvents = await this.getDepositEvents(this.lastProcessedEth1BlockNumber + 1, blockNumber);
+      } catch (ex) {
+        this.logger.warn(`eth1: failed to get deposit events from ${this.lastProcessedEth1BlockNumber + 1}`
+          + ` to ${blockNumber}`);
+        continue;
+      }
+      let success = true;
+      for (const [blockNumber, blockDepositEvents] of getDepositEventsByBlock(rangeDepositEvents)) {
+        if (!await this.processBlock(blockNumber, blockDepositEvents)) {
+          success = false;
+          break;
+        }
+      }
+      // no error, it's safe to update rangeBlockNumber
+      if (success) {
+        rangeBlockNumber = blockNumber;
+        this.lastProcessedEth1BlockNumber = blockNumber;
```

Comment:
we need it because the latest eth1Data block hash may not be `blockNumber`, `blockNumber` may not contain any deposit events. We update `lastProcessedEth1BlockNumber` to `blockNumber` in order not to process old blocks again the next round.